Readme.md

```diff
@@ -386,6 +388,174 @@ export function MyForm({ isNewItem, addItem, updateItem }: Props): JSX.Element =
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        /* Since the FormField could be either a value or an object, the fields property is optional 
+        and you will need to check for existance */
+        value={fields.address.fields?.streetName.value}
+        onChange={(v) =>
+          fields.address.fields?.streetName.onChange(v.target.value)
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+Validations also work with nested objects
+
+```ts
+import { useForm } from 'react-use-form-library';
+
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      address: {
+        streetName: '',
+      },
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+    validations: {
+      address: {
+        streetName: 'required',
+      },
+    }
+  });
+
+  // ...
+```
+
+Since the fields property within FormField is also based on a proxy object, you can access nested properties without
+explicitely declaring them
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {},
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        value={fields.bankDetails.fields?.address.fields?.streetName.value}
+        onChange={(v) =>
+          fields.bankDetails.fields?.address.fields?.streetName.onChange(
+            v.target.value
+          )
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+---
+
+## FieldSet
+
+A FieldSet is basically an array of FormFields. Contrary to nested objects, a FieldSet will only be instantiated if it is **explicitely** declared in the model.
+
+This is because at the moment of field instantiation, the addField method within a Form (or a FormField if it is a nested object) will as if the model value is actually an Array. It is not possible of implicitely determine if the field is an array based only on the type.
+
+The FieldSet class extends Array, so you can access all methods an array has, plus custom methods that work similarly as those in FormField (more in the [API](#fieldset-1))
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      emails: ['example@email.com'],
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      // a FieldSet is iterable
+      {fields.emails.map((field, index) => {
+        // Every FieldSet item is a FormField
+        return (
+          <div className="input">
+            <input
+              value={field.value}
+              onChange={(v) => field.onChange(v.target.value)}
+            />
+            // FieldSet has a helpful removeField method
+            <button onClick={() => fields.emails.remove(index)}>remove</button>
```

Comment:
why do we even need the index here? the `field` object should know its index, so `field.remove()` is a lot easier to call.

Reply:
Is there a more obvious way of achieven this? My first guess was to provide a function on field creation like this:

```
items.forEach((item) => {
      this.fields.push(
        new FormField({
          value: item,
          onUpdate: this.onUpdate,
          validation: this.validation,
          remove: () => {
            this.fields.splice(Math.max(this.fields.length - 1, 0), 1);
          },
        })
      );
    });
```

Reply:
added remove method